### ISSUE TITLE
Configure app domain and port?

### ISSUE THREAD

**Issue by AndrewCraswell** [OfficeDev/teams-toolkit#4205]
It seems like the domain and port are hardcoded to assign to `localhost` and `53000` respectively. But this causes issues for those of us at Microsoft who are developing First Party applications. For example, using `localhost` as an AAD Reply Url is forbidden, and we often get around this by redirecting to a custom domain `<appname>.microsoft.com` using port `443`. Then we can add an entry in our Windows Host files to redirect `https://<appname>.microsoft.com` to `localhost`, and things will work. The alternative is to use something like Ngrok, or obfuscate the `localhost` name behind a vanity Url (https://aka.ms/<appname>). Although the latter example is now a violation of security guidelines as well and is more recently being checked for.

Using Create React App, we set these values in our `.env` file as such:

```
HOST=<appname>.microsoft.com
PORT=443
HTTPS=true
```

Without being able to configure or override the domain and port, many people who have stringent security guidelines won't be able to migrate to Teams Toolkit I suspect.

**Comment by adashen** [comment]
Hi Andrew, thanks for raising the issue https://github.com/OfficeDev/TeamsFx/issues/4205. And we would like to know more about the detail requirements from you. 1. You mentioned "localhost" is not supported, so is it the limitation in 1st party app scenario? 2. And your workaround would be config the "windows host" file to route to localhost. Is the requirement here is to have aad app configured with the desired redirect url? So is it possible for you to workaround this by manually config the aad app in portal?
And currently we are working on some improvements in the toolkit. The major one is we will provide the aad app registration manifest to developer. Which means, developer could customize the app registration in the manifest. So it could be configured with the desired domain. Feel free to let me know whether this scenario fit into your needs.

**Comment by 1iveowl** [comment]
It would be really helpful to be able to set the port. I'm unable to run the tutorial because something is already occupying 53000. Thank you.

**Comment by MuyangAmigo** [comment]
Hey @1iveowl , we have features planned for developers to set custom domain while debugging the application in our roadmap, and will share updates once that is ready for you test out. Please stay tuned and thanks for being patient with us :)

**Comment by ichivers** [comment]
Any progress on this?  I'm trying to debug using android emulator and need to change the hostname to something other than localhost.

**Comment by AndrewCraswell** [comment]
Depending on your project setup, you might be able to override the settings that Teams Toolkit is applying. We use Create-React-App, and so the settings for hostname, port, and https can be controlled using `PORT`, 'HOST', and `HTTPS` respectively. Teams Toolkit is loading their own environment variables which overwrote what we had configured in our `.env` file.

One way around this is to modify the npm start scripts to reload your `.env` file after loading the toolkit environment variables:


```json
scripts: {
  "dev:teamsfx": "env-cmd --silent -f .env.teamsfx.local \"env-cmd --silent -f .env yarn start\"",
  "start": "craco start",
}
```

Now when debugging locally (F5), the `dev:teamsfx` script will run, load the toolkit environment variables (which does contain some variables we need set), then we load our own `.env` file to override the variables we need to control. In the `.env` file we specify these values:

```
PORT=443
HTTPS=true
HOST=<our hostname>.com
```

The downside to this is that we had to add the `env-cmd`  dependency.
